### ISSUE TITLE
Moons with Surface/Biome Issues

### ISSUE THREAD

**Issue by ZBok** [KSP-RO/RealSolarSystem#161]
Hi, thanks so much for what is the greatest mod in KSP. I have almost zero experience here on Github, apologies if I post wrongly or don't include enough detail.

Many moons of Saturn/Jupiter seem to have issues concerning their surface and/or biome map. Since these issues are not consistent, I'll list them each separately. All of these observations were found consistently across a fresh 1.3.1 RSS + Kopernicus only install, as well as 1.6.1 + many mods install and a 1.3.1 + many mods install.

Enceladus: Surface is like, east/west reversed when compared with reality. Linear surface features can be observed to be pointing north west when in reality they point north east, etc. Compare images from wikipedia vs ingame to see what I mean - all the lines are backwards. Can't tell what's going on with the biome map here.

Dione: As with Enceladus, surface features are east-west reversed. Biome map is way off but I can't figure out exactly how.

Rhea: As above, features are east-west reversed. Biome map is... not correlating with the surface.

Titan: Can't see jack. No useful observations.

Tethys + Mimas: surface features are correct relative to reality, but the biome map has been inverted or flipped around the equator.

Io: Has surface features east-west reversed as per Dione/Enceladus/Rhea above.

Europa: Perfect. Matches reality and biome map.

Ganymede: As Io, has east-west reversed. Surface features running east in game that should run west etc

Callisto: Perfect, matches reality and biome map.

Iapetus: This one caused me literal pain. I thought I had it nutted three times before I finally figured it out.
Iapetus surface matches reality, so that's great, but the only way the surface of matches the biome map everywhere (almost), is if you flip the biome map around the equator, like described with Mimas/Tethys. But the red biome (named Turgis) is the exception, it does not actually line up with the Turgis crater, it lines up perfectly with the Engelier crater, so I think it's been either misnamed or misplaced. Once the biome map is inverted around the equator, everything looks great except for that.

Z

I have triple checked these observations as I made a few errors when posting initially, apologies in advance if I have missed any more mistakes.

**Comment by PhineasFreak** [comment]
There are two probable reasons for these orientation issues:

* The fact that all color maps should be flipped horizontally before being exported for RSS uses (example: Earth color map).
* The fact that the biome maps have been incorrectly flipped vertically. Vertical flipping is only required for **.dds** textures and not for other texture formats (including **.bmp**/**.png**/**.tga**).

**Comment by ZBok** [comment]
Yep, they could explain pretty much everything I observed, except for the Turgis biome on Iapetus.

**Comment by ZBok** [comment]
It's probably relevant to include the fact that I'm using 8k textures, I've not tried the others, but if it would be useful to do so please let me know and I will test them all.

I have one small question to ask - if I was to place a permanent base on one of the moons with a currently incorrect surface, like Rhea, and said surface was fixed in an update, would said patch likely delete/blow up/otherwise mess with the surface base?

Follow up question, if the above was done on a moon like Iapetus, which currently has the correct surface but incorrect biome map, would such a surface base have issues if/when a fix is applied?

Definitely not trying to rush a fix, just wondering how to proceed with surface base plans in the meantime.

**Comment by PhineasFreak** [comment]
The texture resolution does not matter. All lower resolution textures are derived from the high resolution ones (and in the case of small bodies they are shared between the LR/MR/HR packs).

Biome/color/normal maps do not touch terrain morphology. Height maps do though and they will probably need to be updated when the rest of the texture orientations are corrected. So yes, when the textures are fixed you are going to have a bad time.

Current problem: nobody has the original textures in hand for the RSS bodies and the sources that RSS provide are old enough that most of them do not exist anymore. CICLOPS does have most of the color maps for the Saturnian moons and some other sources have the Europa/Io color maps but we are missing the Callisto/Ganymede ones.

**Comment by pap1723** [comment]
So is this a texture issue? We cannot use the current textures, convert them out of dds and get it working?

**Comment by PhineasFreak** [comment]
We can use the existing ones but importing/editing/exporting .dds files (lossy compression) would not be ideal.

Texture maps using .png or .bmp formats are not affected (lossless compression).

**Comment by PhineasFreak** [comment]
I have been slowly going through all CBs and fixing stuff were high resolution uncompressed global surface maps are available:

- Io: removed false coloring
- Europa: desaturated the color map by 25%
- Ganymede: no work done yet
- Callisto: no work done yet (if needed)

- Mimas: new color map, added missing normal and height maps
- Enceladus: removed false coloring, added missing normal and height maps, added one biome
- Tethys: work in progress
- Dione: new color map, added missing normal and height maps
- Rhea: work in progress
- Titan: no issues here. Ideally it should get the "Earth" treatment for its lakes (separate lake-only color map for ScaledSpace)
- Iapetus: complete overhaul (new color, normal, height and biome maps)

Additional CBs improved/fixed:

- Deimos & Phobos: removed their color maps completely due to their size (KSP has issues managing the load/unload distance for such small bodies). Color is now managed by PQS (like the stock KSP Gilly)
- Triton: removed false coloring, added missing normal, height and biome maps

I have not checked the Uranus moons (Miranda, Ariel, Umbriel, Titania & Oberon), they may have similar orientation/biome issues. There is not much to work with, most of their surfaces are made up.

**Comment by PhineasFreak** [comment]
Everyhing should be fixed now. All changes are commited for the next RSS assets release.
